docs/features/incremental-generators.cookbook.md

```diff
@@ -0,0 +1,810 @@
+private static string Generate(ClassDeclarationSyntax c)
+{
+    var className = c.Identifier.ToString();
+    var propertyNames = new List<string>();
+    foreach (var member in c.Members)
+    {
+        if (member is PropertyDeclarationSyntax p)
+        {
+            propertyNames.Add(p.Identifier.ToString());
+        }
+    }
+}
+```
+
+This is really all we need. If the serialized values of the properties
+is their string value, the generated code just needs to call `ToString()` on
+them. The only remaining question is what `using`s to put at the top of the file.
+Since our template uses a string builder, we'll need `System.Text` for that, but
+all other types appear to be primitives, so that's all we'll need. Putting it all
+together:
+
+```C#
+private static string Generate(ClassDeclarationSyntax c)
+{
+    var sb = new StringBuilder();
+    int indent = 8;
+    foreach (var member in c.Members)
+    {
+        if (member is PropertyDeclarationSyntax p)
+        {
+            var name = p.Identifier.ToString();
+            appendWithIndent($"addWithIndent($\"\\\"{name}\\\": ");
+            if (p.Type.ToString() != "int")
+            {
+                sb.Append("\\\"");
+            }
+            sb.Append($"{{this.{name}.ToString()}}");
+            if (p.Type.ToString() != "int")
+            {
+                sb.Append("\\\"");
+            }
+            sb.AppendLine(",\");");
+        }
+    }
+
+    return $@"
+using System.Text;
+partial class {c.Identifier.ToString()}
+{{
+    public string Serialize()
+    {{
+        var sb = new StringBuilder();
+        sb.AppendLine(""{{"");
+        int indent = 8;
+
+        // Body
+{sb.ToString()}
+
+        sb.AppendLine(""}}"");
+
+        return sb.ToString();
+
+        void addWithIndent(string s)
+        {{
+            sb.Append(' ', indent);
+            sb.AppendLine(s);
+        }}
+    }}
+}}";
+    void appendWithIndent(string s)
+    {
+        sb.Append(' ', indent);
+        sb.Append(s);
+    }
+}
+```
+
+This ties cleanly into the other serialization examples. By finding all the
+appropriate class declarations in the Compilation's SyntaxTrees and passing
+them to the above Generate method we can build new partial classes for each
+type that opt-ed in to generated serialization. Unlike other technologies,
+this serialization mechanism happens entirely at compile time and can be
+specialized exactly to what was written in the user class.
+
+### Auto interface implementation
+
+TODO:
+
+## Breaking Changes:
+
+* None currently
+
+## Open Issues
+
+This section track other miscellaneous TODO items:
+
+**Framework targets**: May want to mention if we have framework requirements for the generators, e.g. they must target netstandard2.0 or similar.
+
+**Conventions**: (See TODO in [conventions](#conventions) section above). What standard conventions are we suggesting to users?
+
+**Partial methods**: Should we provide a scenario that includes partial methods? Reasons:
```

Comment:
Not sure why this is in open issues, looks like we have a scenario that includes partial methods.